discord/commands/options.py

```diff
@@ -22,14 +22,43 @@
 DEALINGS IN THE SOFTWARE.
 """
 
+from __future__ import annotations
+
 import inspect
-from typing import Any, Dict, List, Literal, Optional, Union
+from typing import TYPE_CHECKING, Dict, List, Literal, Optional, Type, Union
 from enum import Enum
 
-from ..abc import GuildChannel
+from ..abc import GuildChannel, Mentionable
 from ..channel import TextChannel, VoiceChannel, StageChannel, CategoryChannel, Thread
 from ..enums import ChannelType, SlashCommandOptionType, Enum as DiscordEnum
 
+if TYPE_CHECKING:
+    from ..ext.commands import Converter
+    from ..user import User
+    from ..member import Member
+    from ..message import Attachment
+    from ..role import Role
+
+    InputType = Union[
+        Type[str],
```

Comment:
I'm not sure if this applies to the generic `Type` but isn't a type of a class its metaclass? For all classes that don't have custom metaclasses this would be `type`.

Reply:
A `Type[class]` typehint means the value can either be `class` itself or a subclass